surprise/builtin_datasets.py

```
@@ -19,8 +21,12 @@ def get_dataset_dir():
 
     folder = os.environ.get('SURPRISE_DATA_FOLDER', os.path.expanduser('~') +
                             '/.surprise_data/')
-    if not os.path.exists(folder):
+    try:
         os.makedirs(folder)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            # reraise exception if folder exists and creation failed.
```

don't we ignore the exception precisely when the folder exists?

---

@lrebscher ?

---

no need to push @MeTaNoV. Thanks.

---

@NicolasHug, I always push when it's a red BLOCKER! ;)

---

This is open source work. We're not paid for this.

Please reconsider your approach.

---

all GOOD! :)

---

@NicolasHug , happy national day? :dancers: ;)

---

and for the story, I am taking over the legacy of @lrebscher

---

@NicolasHug indeed! 

It should have been:
```suggestion
            # reraise exception if folder does not exist and creation failed.
```